Documentation/raspi-pwm.md

```diff
@@ -0,0 +1,180 @@
+```
+
+You are all setup, the basic example should now work with the PWM and channel you have selected.
+
+### Activating 2 channels
+
+| PWM0 | PWM0 GPIO | PWM0 Function | PWM0 Alt |  PWM1 | PWM1 GPIO | PWM1 Function | PWM1 Alt | dtoverlay |
+| PWM0 | 12 | 4 | Alt0 | PWM1 | 13 | 4 | Alt0 | dtoverlay=pwm-2chan,pin=12,func=4,pin2=13,func2=4 |
+| PWM0 | 18 | 2 | Alt5 | PWM1 | 13 | 4 | Alt0 | dtoverlay=pwm-2chan,pin=18,func=2,pin2=13,func2=4 |
+| PWM0 | 12 | 4 | Alt0 | PWM1 | 19 | 2 | Alt5 | dtoverlay=pwm-2chan,pin=12,func=4,pin2=19,func2=2 |
+| PWM0 | 18 | 2 | Alt5 | PWM1 | 19 | 2 | Alt5 | dtoverlay=pwm-2chan,pin=18,func=2,pin2=19,func2=2 |
+
+Edit the /boot/config.txt file and add the dtoverlay line in the file. You need root privileges for this:
+
+```bash
+sudo nano /boot/config.txt
+```
+
+Save the file with `ctrl + x` then `Y` then `enter`
+
+Then reboot:
+
+```bash
+sudo reboot
+```
+
+You are all setup, the basic example should now work with the PWM and channel you have selected.
+
+## Solving permission issues
+
+When running the basic code, you may have a lack of permissions:
+
+```
+Unhandled exception. System.UnauthorizedAccessException: Access to the path '/sys/class/pwm/pwmchip0/export' is denied.
+ ---> System.IO.IOException: Permission denied
+   --- End of inner exception stack trace ---
+   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter)
+   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode)
+   at System.IO.FileStream.OpenHandle(FileMode mode, FileShare share, FileOptions options)
+   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
+   at System.IO.StreamWriter.ValidateArgsAndOpenPath(String path, Boolean append, Encoding encoding, Int32 bufferSize)
+   at System.IO.StreamWriter..ctor(String path)
+   at System.IO.File.WriteAllText(String path, String contents)
+   at System.Device.Pwm.Channels.UnixPwmChannel.Open()
+   at System.Device.Pwm.Channels.UnixPwmChannel..ctor(Int32 chip, Int32 channel, Int32 frequency, Double dutyCycle)
+   at System.Device.Pwm.PwmChannel.Create(Int32 chip, Int32 channel, Int32 frequency, Double dutyCyclePercentage)
+   at TestTest.Program.Main(String[] args) in C:\tmp\TestTest\TestTest\Program.cs:line 12
+Aborted
+```
+
+You have 2 options: running the code with root privileges or adding your user to the pwm group.
+
+### Running code with root privileges
+
+This is straight forward, you have to use ```sudo``` to run your application. Let's say your application is names ```yourapplication```, once in the same directory as your application, it will then be:
+
+```bash
+sudo ./yourapplication
+```
+
+### Adding your user to the right permission group
+
+You will have to create a [specific group in udev](https://raspberrypi.stackexchange.com/questions/66890/accessing-pwm-module-without-root-permissions).
```

Comment:
This should no longer be necessary. After the latest kernel updates, the lines below are by default part of 99-com.rules. I had asked for this some time ago, and meanwhile it was actually done! See https://github.com/RPi-Distro/raspberrypi-sys-mods/commit/eef0792e899ede22f641e017bae4bdda7e12b9d1

Reply:
It's not because it's in the latest version that it should not be present in this documentation. You may not always use last version. This is something to check in all the cases.

Reply:
True, but maybe explicitly mention that updating the Pi should fix this as well?

Reply:
Fair! Will adjust. Thanks as always for the great insights.